assets/css/query-builder.css

```diff
@@ -602,18 +614,105 @@ margin-bottom: 8em;
 }
 
 
-#search-history tbody button,
-#query-builder .actions button,
-#search-manager .history-controls button {
+.search-history table{
+  table-layout: fixed;
+}
+.search-history tbody button {
 	margin: 0 0.125em;
 	padding: 0;
-	background: transparent;
 }
 
 td.q {
 	cursor: pointer;
 }
 
+th.num,
+td.num{ width: 8em; }
+
+th.history-controls,
+td.history-controls{ width: 114px; }
+
+.search-history > h1,
+.search-history--actions{
+	display: inline-block;  
+}
+
+.search-history--actions{
+	padding-left: 0.4em;
+	position: relative;
+	top: 12px;
+}
+
+.search-history--actions::after{
+	content: "";
+	display: table;
+	clear: both;
+}
+
+.search-history .action{
+  border: 1px solid #aaa;
+  background-color:#f8f8f8;
+  background: linear-gradient(#fff, #eee);
+  font: inherit;
+  height: 32px;
+  line-height: 30px;
+  color: inherit;
+  font-size: 0.85em;
+  font-weight: 600;
+  padding: 0 12px;
+  border-radius: 3px;
+  box-shadow: 0 1px 0 #f8f8f8;
+  margin: 0;
+  float: left;
+}
+
+.search-history .action:hover{
+  border-color: #999;
+  box-shadow: 0 1px 0 #f3f3f3;
+}
+
+.search-history .action:active{
+  background-color:#eee;
+  background: linear-gradient(#eee, #f8f8f8);
+  box-shadow: inset 0 1px 0 rgba(0,0,0,0.08), inset 0 -1px 0 rgba(255,255,255,0.125);
+  color: #888;
+}
+
+.search-history--actions .action-search-history-download{
+	border-radius: 3px 0 0 3px;
+}
+
+.search-history--actions .action-search-history-clear{
+	border-radius: 0 3px 3px 0;
+	border-left: 0;
+}
+
+.search-history caption,
+.search-history .history-controls .label-text{
```

Comment:
We're defining label behaviors at least three different ways, to do the same thing.